### ISSUE TITLE
uninitialized constant Kubeclient::GoogleApplicationDefaultCredentials

### ISSUE THREAD

**Issue by Stono** [abonas/kubeclient#324]
Hey:

```
require 'helpers/tcp'
require 'kubeclient'
require 'googleauth'

RSpec.describe '#at-to-gcp' do                                                                                                                                                             before do
  auth_options = {
        bearer_token: Kubeclient::GoogleApplicationDefaultCredentials.token                                                                                                                     
  }
end
```

Gives me:

```
  1) #at-to-gcp when the echo-server service is running should be able to establish a tcp connection
     Failure/Error: bearer_token: Kubeclient::GoogleApplicationDefaultCredentials.token

     NameError:
       uninitialized constant Kubeclient::GoogleApplicationDefaultCredentials
     # ./spec/at_to_gcp.spec:9:in `block (2 levels) in <top (required)>'
```

Gemfile:

```
❯ cat Gemfile
# frozen_string_literal: true

source "https://rubygems.org"

git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }

gem 'rspec'
gem 'kubeclient'
gem 'googleauth'
```

any ideas?

**Comment by moolitayer** [comment]
As far as I can tell we have no released version with the feature you are using.

@Stono are you looking to have this feature included in a release?

**Comment by moolitayer** [comment]
The README at github reflects master. If you want to view the README for the version you are using it's best to view the one distributed with the gem.

Anyway I understand a new release is required. I will try to get to that tomorrow. Please bug me if you get no response. 
cc @cben

**Comment by Stono** [comment]
Ah that makes sense, thank you :-) I've worked around it for now so no rush:


```
module Helpers
  module Kubernetes
    def cluster(project_id, zone, cluster_id)
      return @cluster if not @cluster.nil?

      scopes = [
        'https://www.googleapis.com/auth/cloud-platform'
      ]
      service = Google::Apis::ContainerV1::ContainerService.new
      service.authorization = Google::Auth.get_application_default(scopes)

      cluster_info = service.get_zone_cluster(project_id, zone, cluster_id).to_h
      master_endpoint = "https://#{cluster_info[:endpoint]}"

      cert_store = OpenSSL::X509::Store.new
      master_ca = cluster_info[:master_auth][:cluster_ca_certificate]
      master_ca = Base64.decode64(master_ca)
      cert_store.add_cert(OpenSSL::X509::Certificate.new(master_ca))

      ssl_options = {
        cert_store: cert_store,
        verify_ssl: OpenSSL::SSL::VERIFY_PEER
      }

      log("Using #{master_endpoint} as Kubernetes host")
      auth_options = {
        bearer_token: service.authorization.access_token
      }

      @cluster = Kubeclient::Client.new(
        master_endpoint,
        'v1',
        auth_options: auth_options,
        ssl_options: ssl_options
      )
      @cluster
    end
  end
end
```

**Comment by cben** [comment]
We just released 3.1.0 that should include this (changelog: #328).  Please test.

**Comment by cben** [comment]
Closing as I believe this is solved — 3.1.* releases contain `Kubeclient::GoogleApplicationDefaultCredentials`.
